src/main/java/teammates/common/util/TimeHelper.java

```diff
@@ -215,20 +215,22 @@ public static String formatTime12H(Date date) {
         return sdf.format(date);
     }
 
-    public static String formatDateTimeForComments(Date date) {
+    public static String formatDateTimeForComments(Date date, double sessionTimeZone) {
         if (date == null) {
             return "";
         }
         SimpleDateFormat sdf = null;
         Calendar c = Calendar.getInstance(SystemParams.TIME_ZONE);
+        TimeZone timeZone = getTimeZoneFromDoubleOffset(sessionTimeZone);
+        c.setTimeZone(timeZone);
         c.setTime(date);
+        sdf = new SimpleDateFormat("EEE, dd MMM yyyy, hh:mm a Z");
+        sdf.setTimeZone(timeZone);
         if (c.get(Calendar.HOUR_OF_DAY) == 12 && c.get(Calendar.MINUTE) == 0) {
-            sdf = new SimpleDateFormat("EEE, dd MMM yyyy, hh:mm");
-            sdf.setTimeZone(SystemParams.TIME_ZONE);
-            return sdf.format(date) + " NOON UTC";
+            sdf = new SimpleDateFormat("EEE, dd MMM yyyy, hh:mm Z");
+            sdf.setTimeZone(timeZone);
+            return sdf.format(date) + " NOON";
```

Comment:
I believe "NOON" should come before the timezone, you can look up https://docs.oracle.com/javase/7/docs/api/java/text/SimpleDateFormat.html for how you can add text in the pattern.